File: TeeNew/Bar_StackGroup.pas
unit Bar_StackGroup;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeCanvas, TeEngine, Series, TeeProcs, Chart;

type
  TBarStackGroup = class(TBaseForm)
    Series1: TBarSeries;
    Series2: TBarSeries;
    Series3: TBarSeries;
    Series4: TBarSeries;
    Series5: TBarSeries;
    Label1: TLabel;
    Edit1: TEdit;
    UpDown1: TUpDown;
    Label2: TLabel;
    procedure Edit1Change(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TBarStackGroup.Edit1Change(Sender: TObject);
var t : Integer;
begin
  inherited;
  for t:=0 to UpDown1.Position-1 do
      (Chart1.Series[t] as TBarSeries).StackGroup:=0 ;
  for t:=UpDown1.Position to Chart1.SeriesCount-1 do
      (Chart1.Series[t] as TBarSeries).StackGroup:=1 ;
end;

procedure TBarStackGroup.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.SeriesList.FillSampleValues(6);
end;

initialization
  RegisterClass(TBarStackGroup);
end.

File: TeeNew/Builder2010/Bar_StackGroup.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Bar_StackGroup.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TBarStackGroup *BarStackGroup;
//---------------------------------------------------------------------------
__fastcall TBarStackGroup::TBarStackGroup(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TBarStackGroup::Edit1Change(TObject *Sender)
{
  for (int t=0; t<UpDown1->Position; t++)
   dynamic_cast<TBarSeries*>(Chart1->Series[t])->StackGroup = 0 ;

  for (int t=UpDown1->Position; t<Chart1->SeriesCount(); t++)
   dynamic_cast<TBarSeries*>(Chart1->Series[t])->StackGroup = 1 ;
}
//---------------------------------------------------------------------------

void __fastcall TBarStackGroup::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(6);
  Series2->FillSampleValues(6);
  Series3->FillSampleValues(6);
  Series4->FillSampleValues(6);
  Series5->FillSampleValues(6);
}
//---------------------------------------------------------------------------

